### PR TITLE
Fix/deferred attribute assignment

### PR DESCRIPTION
**Pull request by jarekwg**
Reverts #367
Adds a test for #331

PR also sneaks in some linter configs.

Issue not fixed here. I suspect it's a django issue. Findings reported in #331.

- [ ] Update `CHANGES.rst` file to describe the changes, and quote according issue with `GH-<issue_number>`.

### CODE REVIEW

**Comment by jarekwg** [comment]
Yeah these failures are expected. Going to try see if the fix can be made in django itself. Just made PR now, to include failing test.
I guess, for the time being, the most useful fix would be #336, but it comes with its own edge cases that it fails to cover...

**Comment by Natim** [comment]
can you add a isort rule to the makefile too?

File: model_utils/tracker.py [hunk 1]
```diff
@@ -40,10 +39,7 @@ def __get__(self, instance, owner):
         if instance is None:
             return self
         was_deferred = self.field_name in instance.get_deferred_fields()
-        try:
-            value = self.descriptor.__get__(instance, owner)
-        except AttributeError:
-            value = self.descriptor
+        value = self.descriptor.__get__(instance, owner)
```
Comment (Natim) [inline model_utils/tracker.py]:
```suggestion
        try:
            value = self.descriptor.__get__(instance, owner)
        except AttributeError:
            value = self.descriptor
```

Comment (jarekwg) [reply model_utils/tracker.py]: Hey, this just reverts my changes?
I've already forgotten a lot of what was going on here, but I believe the try-catch was only a bandaid fix that caused other edge cases to fail.
The real fix, I believe, needs to be made in django. See my open ticket and PR with them here: https://code.djangoproject.com/ticket/30427#ticket
Helping get that merged in would, then allow us to fix the issue properly here.

Comment (Natim) [reply model_utils/tracker.py]: Ok thank you for your advice.

**Comment by jarekwg** [comment]
> can you add a isort rule to the makefile too?

I've not used makefiles this way before, but can take a look when it's time to finally look into getting this merged in. I think we're blocked on django currently.

**Comment by jarekwg** [comment]
Closing. Idk if this was addressed in the end by someone else somewhere else, but I don't have the resources to delve into here again.. ;/
